File: sdk/php/src/Resources/UrlShortener.php
<?php

namespace HiTechCloud\SDK\Resources;

class UrlShortener extends BaseResource
{
    /**
     * Shorten URL
     */
    public function shorten(array $data): array
    {
        return $this->http->post('/api/url-shortener/shorten', $data);
    }

    /**
     * List links
     */
    public function listLinks(): array
    {
        return $this->http->get('/api/url-shortener/links');
    }

    /**
     * Get config
     */
    public function getConfig(): array
    {
        return $this->http->get('/api/url-shortener/config');
    }

    /**
     * Get stats
     */
    public function getStats(): array
    {
        return $this->http->get('/api/url-shortener/stats');
    }

    /**
     * Get link
     */
    public function getLink(int $linkId): array
    {
        return $this->http->get('/api/url-shortener/links/' . $serviceId);
    }

    /**
     * Delete link
     */
    public function deleteLink(int $linkId): array
    {
        return $this->http->delete('/api/url-shortener/links/' . $serviceId);
    }

}

File: sdk/csharp/Resources/UrlShortener.cs
namespace HiTechCloud.SDK.Resources;

public class UrlShortener : BaseResource
{
    public UrlShortener(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> ShortenAsync(Dictionary<string, object> data)
    {
        return await Http.PostAsync("/api/url-shortener/shorten", data);
    }

    public async Task<Dictionary<string, object?>> LinksAsync()
    {
        return await Http.GetAsync("/api/url-shortener/links");
    }

    public async Task<Dictionary<string, object?>> ConfigAsync()
    {
        return await Http.GetAsync("/api/url-shortener/config");
    }

    public async Task<Dictionary<string, object?>> StatsAsync()
    {
        return await Http.GetAsync("/api/url-shortener/stats");
    }

    public async Task<Dictionary<string, object?>> LinkAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/url-shortener/links/{id}", id));
    }

    public async Task<Dictionary<string, object?>> DeleteLinkAsync(int id)
    {
        return await Http.DeleteAsync(string.Format("/api/url-shortener/links/{id}", id));
    }
}
